src/main.ts

```diff
@@ -4,7 +4,10 @@ import { approve } from "./approve";
 
 async function run() {
   const token = core.getInput("github-token", { required: true });
-  await approve(token, github.context);
+  const pr_number: number = JSON.parse(
+    core.getInput("pull-request-number") || "0"
+  );
+  await approve(token, github.context, pr_number);
```

Comment:
As with the change to `approve.ts`, let's switch to camelCase and use `undefined` rather than `0`.

I also think using `parseInt` might be a better fit here – it'll let us detect invalid values more easily and makes the intent a little clearer. Perhaps something like this?

```suggestion
  const prNumber = parseInt(core.getInput("pull-request-number"), 10);
  if (Number.isNaN(prNumber)) {
    await approve(token, github.context);
  } else {
    await approve(token, github.context, prNumber);
  }
```

Reply:
Good suggestion!